docs/design/0011-process-manager.md

```diff
@@ -0,0 +1,133 @@
+This is detailed in this [design proposal](https://github.com/elastic/cloud-on-k8s/blob/master/docs/design/0009-pod-reuse-es-restart.md).
+
+The primary use case for pods reuse was switching for one license type to another with network configuration change. 
+Non-TLS to TLS migration is no more relevant as TLS is now in basic.
+
+Reusing pods may be still relevant, where simply restarting Elasticsearch process can be way faster
+and more efficient than replacing the pod, especially when there's configuration changes 
+(minor configuration changes, plugin changes).
+
+### PID 1 zombie reaping
+
+We want to be sure that the signals are properly handled and the Elasticsearch child processes are correctly reaped 
+when the Elasticsearch process terminates.
+
+## Decision Drivers
+
+### Save CPU and memory resources
+
+Having a keystore sidecar container implies allocating more cpu and memories compared to run the keystore-updater process in the
+Elasticsearch container.
+
+### Init process
+
+Controlling the init process has been identified as interesting for these benefits:
+- handle the PID 1 zombie reaping problem
+- gives better control on what defines "healthy" and "ready" for the ES process (this is defined in the sidecar and 
+we won't have to expose ES urls to k8s for that)
+- makes it easier to perform "nominal restarts" of the ES process without having k8s notice about it
+
+### Full Elasticsearch cluster restart to reuse pods
+
+It must be possible to perform a "full cluster restart" and reuse existing pods.
+
+## Considered Options
+
+Where to run the keystore updater?
+* run it in a sidecar container using the standard Elasticsearch image
+    * ++ easy to serve an API to report the keystore updater status
+    * -- request more cpu and memory to the k8s cluster compared to run the keystore updater in the Elasticsearch container
+* run it inside a process manager injected into the standard Elasticsearch container
+    * ++ easy to serve an API to report the keystore updater status
+    * ~ share cpu and memories with the ES process
+    * ~ we have to be extremely conservative with resource usage
+* run the process manager in a sidecar with [process namespace sharing](https://kubernetes.io/docs/tasks/configure-pod-container/share-process-namespace/#understanding-process-namespace-sharing)
+    * ++ easy to serve an API
+    * -- request more cpu and memory to the k8s cluster compared to run the keystore updater in the Elasticsearch container
+* run it using an off the shelf process manager into the standard Elasticsearch container
+    * -- more processes to run an API, the keystore updater and the ES process
+
+How to perform cluster restart?
+* Destroy the pod and recreate it
+    * -- depending on storage class we might not be able to recreate the pod where the volume resides. Only recovery at this point is manual restore from snapshot.
```

Comment:
I'd like to add a few more details here:

`Considering volumes local to a node: during the interval between the pod being delete and a new pod being scheduled to reuse the same volume, there is no guarantee that no other pod will be scheduled on that node, taking up all resources available on the node, preventing the replacing pod to be scheduled.`